serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/ObjectInspectorUtils.java

```diff
@@ -559,7 +558,6 @@ public static StructField getStandardStructFieldRef(String fieldName,
   public static Field[] getDeclaredNonStaticFields(Class<?> c) {
     Field[] f = c.getDeclaredFields();
     ArrayList<Field> af = new ArrayList<Field>();
```

Comment:
Hey @kokila-19 ,

Saw your [comment](https://github.com/apache/hive/pull/5404#issuecomment-2496801034). Here's some more context on the issue and fix:

Basically the above function called `getDeclaredFields()` [method](https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#getDeclaredFields--) where the order of fields is not defined and can be returned in any order. 
Multiple tests in serde modules were detected which were flaky, i.e. they may fail or pass sometimes based on the assumption that the order of getDeclaredFields will be same each time it is called. 
One of the test case is : `org.apache.hadoop.hive.serde2.objectinspector.TestThriftObjectInspectors.testThriftObjectInspectors`
Others can be found on the ticket. 

Regarding the fix: We used this sorting of Fields based on the getSlotValue so that for a given input, the above function returns the fields in a given sorted order each time it is called. 

I understand that Java 17 has more restrictions on accessing private fields. I also saw that you added --add-opens flag in the main pom.xml file. Basically to fix this, we would want to **get the fields array in a fixed deterministic order**(may be via sorting) rather than relying on the non deterministic behaviour of getDeclaredFields(). We can research more and check how this can be implemented in Java 17(may be adding --add-opens flag to other module or by sorting with some other attribute so that the order .

Reply:
@nikunjagarwal321 I have reverted the commit for now to allow us to proceed with Java 17. I have created a Jira ticket to track this issue: [HIVE-28643](https://issues.apache.org/jira/browse/HIVE-28643). We can follow it up and provide updates there.

Reply:
i don't see it's reverted. sorting was removed in this PR